portal/models/reporting.py

```diff
@@ -82,3 +86,89 @@ def get_reporting_stats():
                         stats['encounters'][interv.description].append(st)
 
     return stats
+
+
+def calculate_days_overdue(user):
+    qb = QuestionnaireBank.most_current_qb(user).questionnaire_bank
+    if not qb:
+        return 0
+    trigger_date = qb.trigger_date(user)
+    overdue = qb.calculated_overdue(trigger_date)
+    return (datetime.utcnow() - overdue).days if overdue else 0
+
+
+def overdue_stats_by_org():
+    overdue_stats = defaultdict(list)
+    for user in User.query.filter_by(active=True):
+        if user.has_role(ROLE.TEST):
+            continue
+        overdue = calculate_days_overdue(user)
+        if overdue > 0:
+            for org in user.organizations:
+                overdue_stats[org].append(overdue)
+    return overdue_stats
+
+
+def generate_overdue_table_html(cutoff_days, overdue_stats, user, top_org):
```

Comment:
This might work better as a jinja template, especially if it gets anymore complicated. Concatenating HTML fragments can make it difficult to get a sense of the overall structure of something and is harder to maintain/extend.

Reply:
Yea, it got bloated as I added onto it. Moving it out to a jinja template now.